Source/WebCore/layout/formattingContexts/grid/GridFormattingContext.cpp

```diff
@@ -112,5 +113,15 @@ void GridFormattingContext::layout(GridLayoutConstraints layoutConstraints)
     GridLayout { *this }.layout(layoutConstraints, unplacedGridItems);
 }
 
+GridFormattingContext::PlacedGridItems GridFormattingContext::constructPlacedGridItems(GridAreas gridAreas) const
+{
+    PlacedGridItems placedGridItems;
+    placedGridItems.reserveInitialCapacity(gridAreas.size());
+    for (auto [ unplacedGridItem, gridAreaLines ] : gridAreas)
+        placedGridItems.constructAndAppend(unplacedGridItem, gridAreaLines);
```

Comment:
Is this copying all the grid items into a new vector?

Reply:
If would be better if we could move the data into the new structure and abandon the old thing. But not needed for this first patch (and perhaps isn't even possible).